gstreamer-sdp/src/s_d_p_media.rs

```diff
@@ -0,0 +1,448 @@
+        copy => |ptr| ffi::gst_sdp_media_copy(mut_override(ptr)),
+        free => |ptr| ffi::gst_sdp_media_free(ptr),
+    }
+}
+
+impl SDPMedia {
+    pub fn add_attribute<'a, P: Into<Option<&'a str>>>(&mut self, key: &str, value: P) -> Result<_, ()> {
+        let value = value.into();
+        let value = value.to_glib_none();
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_media_add_attribute(self.to_glib_none_mut().0, key.to_glib_none().0, value.0))
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn add_bandwidth(&mut self, bwtype: &str, bandwidth: u32) -> Result<_, ()> {
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_media_add_bandwidth(self.to_glib_none_mut().0, bwtype.to_glib_none().0, bandwidth))
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn add_connection(&mut self, nettype: &str, addrtype: &str, address: &str, ttl: u32, addr_number: u32) -> Result<_, ()> {
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_media_add_connection(self.to_glib_none_mut().0, nettype.to_glib_none().0, addrtype.to_glib_none().0, address.to_glib_none().0, ttl, addr_number))
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn add_format(&mut self, format: &str) -> Result<_, ()> {
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_media_add_format(self.to_glib_none_mut().0, format.to_glib_none().0))
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn as_text(&self) -> Option<String> {
+        unsafe {
+            from_glib_full(ffi::gst_sdp_media_as_text(self.to_glib_none().0))
+        }
+    }
+
+    pub fn attributes_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_attributes_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn attributes_to_caps(&self, caps: &gst::Caps) -> Result<_, ()> {
+        let result = unsafe {
+            from_glib(ffi::gst_sdp_media_attributes_to_caps(self.to_glib_none().0, caps.to_glib_none().0))
+        };
+        match result {
+            SDPResult::ok => Some(_),
+            _ => Err(()),
+            }
+        }
+    }
+
+    pub fn bandwidths_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_bandwidths_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn connections_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_connections_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn formats_len(&self) -> u32 {
+        unsafe {
+            ffi::gst_sdp_media_formats_len(self.to_glib_none().0)
+        }
+    }
+
+    pub fn get_attribute(&self, idx: u32) -> Option<SDPAttribute> {
```

Comment:
This does not give you ownership of the return value but borrows it to you. It should return a `Option<&SDPAttribute>`. Same for all the other getters